rules/Composer/Application/FileProcessor/ComposerFileProcessor.php

```diff
@@ -45,14 +45,14 @@ public function __construct(
     /**
      * @param File[] $files
      */
-    public function process(array $files): void
+    public function process(array $files, callable $advanceProgressBar): void
```

Comment:
`callable` is very poor contract that would shoot you back with confusing error message. In general, we should code against interface.